sql/src/main/java/org/opensearch/sql/sql/domain/SQLQueryRequest.java

```diff
@@ -148,4 +149,22 @@ private boolean shouldSanitize(Map<String, String> params) {
     }
     return true;
   }
+
+  /** A new toString() with anonymizer parameter to anonymize its query statement. */
+  public String toString(Function<String, String> anonymizer) {
```

Comment:
Just wonder do you see future requirement for different anonymizer? Is `toString(boolean anonymized)` or copy to a new request clear enough?